gcc/rust/util/rust-attributes.cc

```diff
@@ -623,6 +624,12 @@ AttributeChecker::visit (AST::Function &fun)
 		     name);
   };
 
+  auto check_function_name = [] (const char *name, const Identifier &ident) {
+    if (!is_ascii_only (ident.as_string ()))
+      rust_error_at (ident.get_locus (),
+		     "the %<#[%s]%> attribute requires ASCII identifier", name);
+  };
```

Comment:
will this function be used for anything other than the `#[no_mangle]` attribute? If no, we can skip having the `const char *name` argument. I think it would also be cleaner to have it as a `static void` function in the file rather than a lambda, as C++ lambdas are not super easy to read

Reply:
Okay. I will fix it later.